run/data/data.go

```diff
@@ -68,52 +67,52 @@ func (t *Table) HasPrimaryKey() bool {
-	Nullable                            bool                      // true if the column is not NON NULL
-	HasDefault                          bool                      // true if the column has a default
-	IsPrimaryKey                        bool                      // true if the column is a primary key
-	IsForeignKey                        bool                      // true if the column is a foreign key
-	IsForeignKeyReference               bool                      // true if the column is referenced by a foreign key
-	ForeignColumn                       *ForeignColumn            // foreign key database definition
-	ForeignKeyReferences                []string                  // all database names of foreign keys referencing this column
-	ForeignColumnsByForeignKeyReference map[string]*ForeignColumn // all columns referring to this column
-	Orig                                interface{}               `yaml:"-" json:"-"` // the raw database column data
+	Table              *Table                       `yaml:"-" json:"-"` // the table this column is in
+	Name               string                       // the converted name of the column
+	DBName             string                       // the original name of the column in the DB
+	Type               string                       // the converted name of the type
+	DBType             string                       // the original type of the column in the DB
+	IsArray            bool                         // true if the column type is an array
+	Length             int                          // non-zero if the type has a length (e.g. varchar[16])
+	UserDefined        bool                         // true if the type is user-defined
+	Nullable           bool                         // true if the column is not NON NULL
+	HasDefault         bool                         // true if the column has a default
+	IsPrimaryKey       bool                         // true if the column is a primary key
+	IsFK               bool                         // true if the column is a foreign key
+	HasFKRef           bool                         // true if the column is referenced by a foreign key
+	FKColumn           *ForeignKeyColumn            // foreign key column definition
+	FKColumnRefs       ForeignKeyColumns            // all foreign key columns referencing this column
+	FKColumnRefsByName map[string]*ForeignKeyColumn `yaml:"-" json:"-"` // all foreign key columns referencing this column by foreign key name
+	Orig               interface{}                  `yaml:"-" json:"-"` // the raw database column data
 }
 
-type ForeignTable struct {
-	Name             string
-	TableName        string
-	ForeignTableName string
-	Table            *Table `yaml:"-" json:"-"`
-	ForeignTable     *Table `yaml:"-" json:"-"`
+// Foreign Key contains the
+type ForeignKey struct {
+	DBName         string            // the original name of the foreign key constraint in the db
```

Comment:
Can you add the converted name here too?  Just on the off chance someone want to use it for something.

Reply:
And a Names() function on the slice of them.